pyat/at/lattice/cavity_access.py

```diff
@@ -147,137 +162,185 @@ def get_rf_timelag(ring: Lattice, **kwargs) -> float:
     """Set the RF time lag
 
     Parameters:
         ring:           Lattice description
-        timelag:        RF time lag
+        timelag:        RF time lag, expressed as path lengthening [m]
 
     Keyword Arguments:
-        cavpts=None:    Cavity location. If None, look for ring.cavpts,
-                          otherwise take all cavities.
-        array=False:    If False, return the frequency of the selected cavities
-                          with the lowest frequency.
-
-                          If True, return the frequency of all selected cavities
-        copy=False:     If True, returns a shallow copy of ``ring`` with new
-                          cavity elements. Otherwise, modify ``ring`` in-place.
+        cavpts (Refpts | None):  If :py:obj:`None`, look for :pycode:`ring.cavpts`, or
+          otherwise take all cavities.
+        array (bool):       If :py:obj:`False` (default), *timelag* is applied to the
+          main cavities. The timelag of the other cavities is shifted by the same
+          amount.
+
+          If :py:obj:`True`, directly apply *timelag* to the selected cavities. The
+          value must be broadcastable to the number of cavities.
+        copy (bool):    If :py:obj:`True`, returns a shallow copy of *ring* with new
+          cavity elements. Otherwise (default), modify *ring* in-place.
     """
     return set_cavity(ring, TimeLag=timelag, **kwargs)
 
 
 # noinspection PyPep8Naming
-def set_cavity(ring: Lattice, Voltage: Optional[float] = None,
-               Frequency: Optional[float] = None,
-               TimeLag: Optional[float] = None,
-               cavpts: Optional[Refpts] = None,
-               copy: Optional[bool] = False,
-               array: Optional[bool] = False):
+def set_cavity(
+    ring: Lattice,
+    Voltage: float | None = None,
+    Frequency: float | Frf | None = None,
+    TimeLag: float | None = None,
+    cavpts: Refpts | None = None,
+    copy: bool = False,
+    array: bool = False,
+    **kwargs,
+) -> Lattice | None:
     """
     Set the parameters of the RF cavities
 
     Parameters:
-        ring:       lattice description
-        Frequency:  RF frequency [Hz]
-        Voltage:    RF voltage [V]
-        TimeLag:    RF time shift [-ct]
-        cavpts:     Cavity location. If None, look for ring.cavpts, or
-                      otherwise take all cavities
-        array:      If False, the value is applied as described for
-                      set_rf_voltage, set_rf_timelag and set_rf_frequency
-                      If True, directly apply the value to the selected
-                      cavities. The value must be broadcastable to the number
-                      of cavities.
-        copy:       If True, returns a shallow copy of ring with new
-                      cavity elements. Otherwise, modify ring in-place
+        ring:           lattice description
+        Frequency:      RF frequency [Hz]
+        Voltage:        Total RF voltage [V]
```

Comment:
A word on Frf.NOMINAL would be useful here

Reply:
done